include/solvers/mpm_base.h

```diff
@@ -24,6 +24,13 @@
 
 namespace mpm {
 
+//! Variable type
+//! Scalar: boolean, unsigned, int, double
+//! Vector: Vector of size 3
+//! Tensor: Symmetric tensor arranged in voigt notation
+enum class VariableType { Scalar, Vector, Tensor };
+extern std::map<std::string, VariableType> variables;
```

Comment:
What is the reasoning behind defining the variable outside the class?

Reply:
Defining inside the class gives better control of what variable can be listed here

Reply:
I tried doing that, but the test won't compile for multiple definitions.
![image](https://user-images.githubusercontent.com/37140224/89476924-c919a000-d740-11ea-8bf6-b1d66671b2dd.png)